packages/client/hmi-client/src/services/data.ts

```diff
@@ -498,24 +498,18 @@ const fetchResource = async (
 	// eslint-disable-next-line no-async-promise-executor
 	new Promise<FullSearchResults>(async (resolve, reject) => {
 		try {
-			switch (resourceType) {
-				case ResourceType.XDD: // XDD
-					resolve({
-						allData: await searchXDDArticles(term, searchParam?.xdd),
-						allDataFilteredWithFacets: await searchXDDArticles(
-							term,
-							searchParamWithFacetFilters?.xdd
-						)
-					});
-					break;
-				case ResourceType.MODEL: // Models
-					resolve(getAssets(term, ResourceType.MODEL, searchParamWithFacetFilters?.model));
-					break;
-				case ResourceType.DATASET: // Datasets
-					resolve(getAssets(term, ResourceType.DATASET, searchParamWithFacetFilters?.dataset));
-					break;
-				default:
-					break;
+			if (resourceType === ResourceType.XDD) {
+				resolve({
+					allData: await searchXDDArticles(term, searchParam?.[ResourceType.XDD]),
+					allDataFilteredWithFacets: await searchXDDArticles(
+						term,
+						searchParamWithFacetFilters?.[ResourceType.XDD]
+					)
+				});
+			} else if (resourceType === ResourceType.MODEL || resourceType === ResourceType.DATASET) {
+				resolve(
+					getAssets(term, resourceType, searchParamWithFacetFilters?.[resourceType as ResourceType])
+				);
```

Comment:
Why removing the `switch`?

Reply:
I wanted to use the [resourceType] as the attribute to combine the MODEL and DATASET cases. Can't write a case after case though due to the no-fallthrough eslint rule.